CIScripts/Test/Tests/vRouterAgentService.Tests.ps1

```diff
@@ -0,0 +1,119 @@
+Param (
+    [Parameter(Mandatory=$true)] [string] $TestbedAddr
+)
+
+. $PSScriptRoot\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\..\Common\Aliases.ps1
+. $PSScriptRoot\..\..\Common\VMUtils.ps1
+. $PSScriptRoot\..\PesterHelpers\PesterHelpers.ps1
+
+. $PSScriptRoot\..\GetTestConfigurationCodiLegacy.ps1
+$TestConf = Get-TestConfiguration
+$Session = New-PSSession -ComputerName $TestbedAddr -Credential (Get-VMCreds)
+
+Describe "vRouter Agent service" {
+    Context "enabling" {
+        It "is enabled" {
+            Get-AgentServiceStatus -Session $Session `
+                | Should Be "Running"
+        }
+
+        BeforeEach {
+            Enable-AgentService -Session $Session
+        }
+    }
+    
+    Context "disabling" {
+        It "is disabled" {
+            Get-AgentServiceStatus -Session $Session `
+                | Should Be "Stopped"
+        }
+
+        It "does not restart" {
+            Consistently {
+                Get-AgentServiceStatus -Session $Session | Should Be "Stopped"
+            }
+        }
+
+        BeforeEach {
+            Enable-AgentService -Session $Session
+            Disable-AgentService -Session $Session
+        }
+    }
+
+    Context "given vRouter Forwarding Extension is NOT running" {
+        It "crashes" {
+            Eventually {
+                Read-SyslogForAgentCrash -Session $Session -After $BeforeCrash `
+                    | Should Not BeNullOrEmpty
+            } -Duration 60
+        }
+
+        BeforeEach {
+            Disable-VRouterExtension -Session $Session `
+                -AdapterName $TestConf.AdapterName `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName
+            Test-IsVRouterExtensionEnabled -Session $Session `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName `
+                | Should Be $false
```

Comment:
Is using `Should` outside `It` a good practice?

Reply:
This was copied over from old powershell script. I'll remove it.